source/includes/fundamentals/code-snippets/CRUD/delete.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	var uri string
+	if uri = os.Getenv("MONGODB_URI"); uri == "" {
+		log.Fatal("You must set your `MONGODB_URI' environmental variable. See\n\t https://docs.mongodb.com/drivers/go/current/usage-examples/")
+	}
+
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			panic(err)
+		}
+	}()
+
+	client.Database("tea").Collection("ratings").Drop(context.TODO())
+
+	// begin insertDocs
+	coll := client.Database("tea").Collection("ratings")
+	docs := []interface{}{
+		bson.D{{"type", "Masala"}, {"rating", 10}},
+		bson.D{{"type", "Earl Grey"}, {"rating", 7}},
+		bson.D{{"type", "Oolong"}, {"rating", 10}},
+		bson.D{{"type", "Assam"}, {"rating", 7}},
+	}
+
+	result, insertErr := coll.InsertMany(context.TODO(), docs)
+	if insertErr != nil {
+		panic(insertErr)
+	}
+	// end insertDocs
+	fmt.Printf("Number of documents inserted: %d\n", len(result.InsertedIDs))
+
+	// begin deleteMany
+	deleteManyFilter := bson.D{{"rating", bson.D{{"$gt", 8}}}}
+	deleteOptions := options.Delete().SetHint(bson.D{{"_id", 1}})
+
+	deleteManyResult, deleteManyErr := coll.DeleteMany(context.TODO(), deleteManyFilter, deleteOptions)
+	fmt.Printf("Number of documents deleted: %d\n", deleteManyResult.DeletedCount)
```

Comment:
Looks like this `Printf` is inside the example, but the previous `Printf` on L47 is outside the example. Not sure if this was intentional.

Reply:
I choose to exclude L47 since it wasn't the purpose of that section, but I was unsure if it should be included or not. (When I didn't include that line, I got an error saying the cursor wasn't used.)

Reply:
I think the way you have it is actually fine. Including L54 in the `deleteMany` example seems important, as it shows users how to access the `DeletedCount` field. L47 only shows how to count the number of inserted documents which, as you pointed out, is not the purpose of the section. No need to change.